examples/model_selection/plot_cv_indices.py

```diff
@@ -0,0 +1,147 @@
+# and 10 "groups" split unevenly across datapoints.
+#
+# As we'll see, some cross-validation objects do specific things with
+# labeled data, others behave differently with grouped data, and others
+# do not use this information.
+#
+# To begin, we'll visualize our data.
+
+# Generate the label/group data
+n_points = 100
+X = np.random.randn(100, 10)
+
+percentiles_labels = [.1, .3, .6]
+y = np.hstack([[ii] * int(100 * perc)
+               for ii, perc in enumerate(percentiles_labels)])
+
+# Evenly spaced groups repeated once
+groups = np.hstack([[ii] * 10 for ii in range(10)])
+
+
+def visualize_groups(labels, groups, name):
+    # Visualize dataset groups
+    fig, ax = plt.subplots()
+    ax.scatter(range(len(groups)),  [.5] * len(groups), c=groups, marker='_',
+               lw=50, cmap=cmap_data)
+    ax.scatter(range(len(groups)),  [3.5] * len(groups), c=labels, marker='_',
+               lw=50, cmap=cmap_data)
+    ax.set(ylim=[-1, 5], yticks=[.5, 3.5],
+           yticklabels=['Data groups', 'Data label'], xlabel="Sample index")
+
+
+visualize_groups(y, groups, 'no groups')
+
+###############################################################################
+# Define a function to visualize cross-validation behavior
+# --------------------------------------------------------
+#
+# We'll define a function that lets us visualize the behavior of each
+# cross-validation object. We'll perform 5 splits of the data. On each
+# split, we'll visualize the indices chosen for the training set
+# (in blue) and the test set (in red).
+
+
+def plot_cv_indices(cv, X, y, group, ax, n_splits, lw=10):
+    """Create a sample plot for indices of a cross-validation object."""
+
+    # Generate the training/testing visualizations for each CV split
+    for ii, (tr, tt) in enumerate(cv.split(X=X, y=y, groups=group)):
+        # Fill in indices with the training/test groups
+        indices = np.array([np.nan] * len(X))
+        indices[tt] = 1
+        indices[tr] = 0
+
+        # Visualize the results
+        ax.scatter(range(len(indices)), [ii + .5] * len(indices),
+                   c=indices, marker='_', lw=lw, cmap=cmap_cv,
+                   vmin=-.2, vmax=1.2)
+
+    # Plot the data labels and groups at the end
+    ax.scatter(range(len(X)), [ii + 1.5] * len(X),
+               c=y, marker='_', lw=lw, cmap=cmap_data)
+
+    ax.scatter(range(len(X)), [ii + 2.5] * len(X),
+               c=group, marker='_', lw=lw, cmap=cmap_data)
+
+    # Formatting
+    yticklabels = list(range(n_splits)) + ['labels', 'groups']
+    ax.set(yticks=np.arange(n_splits+2) + .5, yticklabels=yticklabels,
+           xlabel='Sample index', ylabel="CV iteration",
+           ylim=[n_splits+2.2, -.2], xlim=[0, 100],
+           title='{}'.format(type(cv).__name__))
+    return ax
+
+
+###############################################################################
+# Let's see how it looks for the `KFold` cross-validation object:
+
+fig, ax = plt.subplots()
+n_splits = 4
```

Comment:
n_splits is also used in other classes, so maybe it's better to move it above

Reply:
ok, comments addressed! re: the legend, if other folks want to remove it then I can do so. One reason to do this was to allow other parts of the documentation to embed the figures one-by-one (e.g. the cross-validation page). But if folks don't think that'll be used, I agree we could remove the legend to save some space